app/models/integration.rb

```diff
@@ -0,0 +1,19 @@
+class Integration < ActiveRecord::Base
+  belongs_to :website
+  has_one    :application
+  validates :name, presence: true
+  validates :provider, presence: true
+
+  def driver
+    Integrations.create(self)
+  end
+
+  def assign_configuration(auth_hash)
+    self.configuration = driver.build_configuration(auth_hash)
+    self.provider = driver.type
+  end
+
+  def get_applications
+    driver.driver.applications
```

Comment:
you can delegate applications to the integration class. That way you can do `driver.applications` here.